src/solvers/smt2/smt2_tokenizer.h

```diff
@@ -23,7 +21,7 @@ class smt2_tokenizert
     line_no=1;
   }
 
-  class smt2_errort : public cprover_exception_baset
```

Comment:
Is there a reason this was removed from the inheritance chain? Would it be better if `message` was removed and substituted with `reason`?

Reply:
In this class, `message` is a `std::ostringstream`, making it really different from the parent class. Really, there is no meaningful commonality.

Reply:
The only benefit would be to catch all these exceptions at once, which could be done by another 'base class with reason' above the existing base, but that's minor IMO.